src/lib.rs

```diff
@@ -325,6 +333,22 @@ where
             .illumina_r1_trim_length(processor.illumina_r1_trim_length())
             .illumina_r2_trim_length(processor.illumina_r2_trim_length())
             .subsample_rate(processor.read_subsample_rate());
+
+        let read_pair_iter = AnyReadPairIter::Direct(read_pair_iter);
+        Ok(FastqProcessorIter {
+            read_pair_iter,
+            processor,
+        })
+    }
+
+    pub fn new_background(processor: &'a Processor, readahead: usize) -> Result<Self, Error> {
+        let read_pair_iter = ReadPairIter::from_fastq_files(&processor.fastq_files())?
+            .illumina_r1_trim_length(processor.illumina_r1_trim_length())
+            .illumina_r2_trim_length(processor.illumina_r2_trim_length())
+            .subsample_rate(processor.read_subsample_rate());
```

Comment:
Would be good to refactor this to a function to create `ReadPairIter` from `&Processor`, so that would not go out of sync with the sinppet above accidentally.